edx_proctoring/tests/test_api.py

```diff
@@ -498,6 +499,150 @@ def test_remove_allowance_for_user(self):
+        )
+        student_allowance_exam2 = ProctoredExamStudentAllowance.get_allowance_for_user(
+            exam_list[1], user_list[1], self.key
+        )
+    
+        all_course_allowances = get_allowances_for_course(self.course_id)
+        self.assertEqual(len(all_course_allowances), 6)
+        self.assertIsNotNone(student_allowance)
+        self.assertEqual(student_allowance.value, '10')
+        self.assertEqual(student_allowance_exam2.value, '45')
+
+    def test_add_same_user_bulk_allowance(self):
+        """
+        Test to add bulk allowances.
+        """
+        user_list = self.create_batch_users(3)
+        user_list.append(user_list[1])
+        exam_list = []
+        exam_list.extend(
+            (create_exam(
+            course_id=self.course_id,
+            content_id="1st exam",
+            exam_name="1st exam",
+            time_limit_mins=self.default_time_limit,
+            is_practice_exam=False,
+            is_proctored=True
+            ),
+            create_exam(
+            course_id=self.course_id,
+            content_id="2nd exam",
+            exam_name="2nd exam",
+            time_limit_mins=90,
+            is_practice_exam=False,
+            is_proctored=True
+            ))
+        )
+        add_bulk_allowances(exam_list, user_list, 'time_multiplier', '1.5')
+        student_allowance = ProctoredExamStudentAllowance.get_allowance_for_user(
+            exam_list[0], user_list[1], self.key
+        )
+        all_course_allowances = get_allowances_for_course(self.course_id)
+        self.assertEqual(len(all_course_allowances), 6)
+        self.assertIsNotNone(student_allowance)
+        self.assertEqual(len(user_list), 4)
+
+    def test_add_same_exam_bulk_allowance(self):
+        """
+        Test to add bulk allowances.
+        """
+        user_list = self.create_batch_users(3)
+        user_list.append(user_list[1])
+        exam_list = []
+        exam_list.extend(
+            (create_exam(
+            course_id=self.course_id,
+            content_id="1st exam",
+            exam_name="1st exam",
+            time_limit_mins=self.default_time_limit,
+            is_practice_exam=False,
+            is_proctored=True
+            ),
+            create_exam(
+            course_id=self.course_id,
+            content_id="2nd exam",
+            exam_name="2nd exam",
+            time_limit_mins=90,
+            is_practice_exam=False,
+            is_proctored=True
+            ))
+        )
+        exam_list.append(exam_list[0])
+        add_bulk_allowances(exam_list, user_list, 'time_multiplier', '1.5')
+        student_allowance = ProctoredExamStudentAllowance.get_allowance_for_user(
+            exam_list[0], user_list[1], self.key
+        )
+        all_course_allowances = get_allowances_for_course(self.course_id)
+        self.assertEqual(len(all_course_allowances), 6)
+        self.assertIsNotNone(student_allowance)
+        self.assertEqual(len(exam_list), 3)
+
```

Comment:
Please add unit tests to handle edge cases. For example:
- Courses without exams
- Exams without time_limit specified
- Multiplier somehow is negative.